File: src/leetcode75/ValidAnagram.java
public class ValidAnagram {
    public static boolean isAnagram(String s, String t) {
        s = s.toLowerCase();
        t = t.toLowerCase();

        int[] charCount = new int[26];
        
        for (char c : s.toCharArray()) {
            charCount[c - 'a']++;
        }

        for (char c : t.toCharArray()) {
            charCount[c - 'a']--;
        }

        for (int count : charCount) {
            if (count != 0) return false;
        }

        return true;
    }

    public static void main(String[] args) {
        String test1s = "anagram";
        String test1t = "nagaram";
        String test2s = "rat";
        String test2t = "car";

        System.out.println(isAnagram(test1s, test1t));
        System.out.println(isAnagram(test2s, test2t));
    }
}

File: src/main/python/validanagram.py
class Solution(object):
    def is_anagram(self, s1, s2):
        if len(s1) != len(s2):
            return False
        d = {}
        for c in s1:
            if c in d:
                d[c] += 1
            else:
                d[c] = 1
        for c in s2:
            if c in d:
                d[c] -= 1
            else:
                d[c] = -1
        for v in d.values():
            if v != 0:
                return False
        return True

if __name__ == '__main__':
    s = Solution()
    print(s.is_anagram('anagram', 'nagaram'))
    print(s.is_anagram('rat', 'car'))
    
